pkg/webui/console/store/middleware/logics/gateways.js

```diff
@@ -201,6 +206,10 @@ const updateGatewayStatisticsLogic = createRequestLogic({
 
     const stats = await api.gateway.stats(id)
 
+    if (!Boolean(stats.last_status_received_at)) {
+      stats.last_status_received_at = new Date()
```

Comment:
Is this the correct assumption? Why would this prop be omitted from the response? In the proptypes it is also set as required.

Reply:
> Why would this prop be omitted from the response?

some gateways don’t send status at all. had discussion about this with @johanstokking

Reply:
I see. Should we set `new Date()` only for heartbeat-triggered updates then?

Reply:
not sure what you mean

Reply:
I mean that for gateways that don't send status, we would then still bump the last-seen in a 60s interval. Yet, we only safely know the gateway was seen when we received `gs.status.receive`, `gs.down.send`, `gs.up.receive`.

Reply:
@johanstokking

Reply:
"Last seen" refers specifically and only to:

- `last_{status|uplink}_received_at`
- `gs.{status|up}.receive`

It should not be updated when fetching stats on the interval. If the gateway is connected but there are no status or uplinks received, it should not be implied that we've seen traffic (this diff). Still, the gateway shows connected.

Maybe "last seen" is the wrong term as it's not just that it's connected, but more "last message received"